test/framework/modules.py

```diff
@@ -750,6 +750,110 @@ def test_exit_code_check(self):
             error_pattern = "Unable to locate a modulefile for 'nosuchmoduleavailableanywhere'"
         self.assertErrorRegex(EasyBuildError, error_pattern, self.modtool.load, ['nosuchmoduleavailableanywhere'])
 
+    def test_check_loaded_modules(self):
+        """Test check_loaded_modules method."""
+        # try and make sure we start with a clean slate
+        self.modtool.purge()
+
+        def check_loaded_modules():
+            "Helper function to run check_loaded_modules and check on stdout/stderr."
+            # there should be no errors/warnings by default is no (EasyBuild-generated) modules are loaded
+            self.mock_stdout(True)
+            self.mock_stderr(True)
+            self.modtool.check_loaded_modules()
+            stdout, stderr = self.get_stdout(), self.get_stderr()
+            self.mock_stdout(False)
+            self.mock_stderr(False)
+            self.assertEqual(stdout, '')
+            return stderr.strip()
+
+
+        # by default, having an EasyBuild module loaded is allowed
+        self.modtool.load(['EasyBuild/fake'])
+
+        # no output to stderr (no warnings/errors)
+        self.assertEqual(check_loaded_modules(), '')
+
+        self.modtool.unload(['EasyBuild/fake'])
+
+        # load OpenMPI module, which also loads GCC & hwloc
+        self.modtool.load(['OpenMPI/1.6.4-GCC-4.6.4'])
+
+        # default action is to print a clear warning message
+        stderr = check_loaded_modules()
+        patterns = [
+            r"^WARNING: Found one or more non-ignored loaded \(EasyBuild-generated\) modules in current environment:",
+            r"^\* GCC/4.6.4",
+            r"^\* hwloc/1.6.2-GCC-4.6.4",
+            r"^\* OpenMPI/1.6.4-GCC-4.6.4",
+            "This is not recommended since it may affect the installation procedure\(s\) performed by EasyBuild.",
+            "To make EasyBuild allow particular loaded modules, use the --allow-loaded-modules configuration option.",
+            "To specify action to take when loaded modules are detected, use "
+                "--detect-loaded-modules={error,ignore,purge,unload,warn}",
+        ]
+        for pattern in patterns:
```

Comment:
Only when one of the patterns isn't found, this will produce output (as a failing test).

The check is an assertion, so you'll see something like:

```
AssertionError: Pattern ... found in ...
```

So, the assertion is that a certain pattern should be found, and if not, the assertion fails and you get an assertion error.

This is done consistently across all tests. It looks backwards, but it's not if you interpret it right.

I could make this one big pattern, sure, but that makes spotting a mismatch visually a lot harder, so I prefer to break it down in parts. Speedup of the tests is going to be negligible, we should optimise for human effort here.